tests/test_kv_config_store.py

```
@@ -128,15 +143,306 @@ class _UnavailableConfigBackend:
+        self.last_set = (category, key, value)
+
+
+def test_migration_re_checks_immediately_before_writing_to_narrow_the_race(
+) -> None:
+    """A write landing between the absence-check and the legacy-value read survives.
+
+    Proves the re-check added directly above the ``set`` call actually does
+    something: without it, this exact interleaving would silently overwrite
+    the operator's real value with the stale legacy one.
+    """
+    legacy_category, (replacement_category, config_keys) = next(
```

Ruff (0.16.3)

[warning] Unpacked variable `legacy_category` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)